src/components/editor/resourceProfileMapping/ResourceProfileMapping.tsx

```diff
@@ -5,42 +5,113 @@ import {
   IStructureDefinition,
   IElementDefinition
 } from "@ahryman40k/ts-fhir-types/lib/R4";
+import useStyles from "./style";
+import { TreeView, TreeItem } from "@material-ui/lab";
+import {
+  ArrowRight,
+  ArrowDropDown,
+  Folder,
+  LocalOffer,
+  Settings
+} from "@material-ui/icons";
+import { Typography } from "@material-ui/core";
 
 type ResourceProfileMappingProps = {
   profile: IStructureDefinition | null;
 };
 
+interface RenderTree {
+  id: string;
+  name: string;
+  children?: RenderTree[];
+}
+
 const ResourceProfileMapping: React.FC<ResourceProfileMappingProps> = ({
   profile
 }) => {
   const dispatch = useDispatch();
   const attributes: Array<IElementDefinition> | undefined =
     profile?.snapshot?.element;
-  const structuredefSettings: string = "structuredefSettings" as string;
-
-  const renderTree: JSX.Element[] | undefined = attributes?.map(
-    (attribute: IElementDefinition, index: number) => {
-      return (
-        <div
-          key={index}
-          onClick={() => {
-            dispatch(selectAttributeId(attribute?.id));
-          }}
-        >
-          {attribute.id}
-        </div>
-      );
-    }
+  const structuredefSettings: string = "structuredefSettings";
+  const classes = useStyles();
+
+  let attributesToRender: RenderTree[] = [];
+  if (attributes) {
+    attributes.forEach((attribute) => {
+      if (attribute.path) {
+        attribute.path.split(".").reduce(
+          (render: any, name: string) => {
+            let node = (render.children = render.children || []).find(
+              (item: RenderTree) => item.name === name
+            );
+            if (!node && attribute.path) {
+              render.children.push({ name, id: attribute.path });
+            }
+            return node;
+          },
+          { children: attributesToRender }
```

Comment:
I want to re-write this with you tomorrow

Reply:
And i'd like to be there too if that's okay